__init__.py

```diff
@@ -12,52 +12,71 @@
 from threading import Timer, Lock
 from uuid import uuid4
-from requests import HTTPError
 
 from adapt.intent import IntentBuilder
 
+import mycroft.audio
 from mycroft.api import DeviceApi, is_paired, check_remote_pairing
 from mycroft.identity import IdentityManager
 from mycroft.messagebus.message import Message
 from mycroft.skills.core import MycroftSkill, intent_handler
-import mycroft.audio
 
 
-PLATFORMS_WITH_BUTTON = ('mycroft_mark_1')
+ACTION_BUTTON_PLATFORMS = ('mycroft_mark_1', 'mycroft_mark_2')
+MAX_PAIRING_CODE_RETRIES = 30
+ACTIVATION_POLL_FREQUENCY = 10  # secs between checking server for activation
 
 
-class PairingSkill(MycroftSkill):
+def _stop_speaking():
+    """Stop speaking the pairing code if it is still being spoken."""
+    if mycroft.audio.is_speaking():
+        mycroft.audio.stop_speaking()
 
-    poll_frequency = 10  # secs between checking server for activation
 
+class PairingSkill(MycroftSkill):
+    """Device pairing logic."""
     def __init__(self):
         super(PairingSkill, self).__init__("PairingSkill")
         self.api = DeviceApi()
-        self.data = None
-        self.time_code_expires = None
+        self.pairing_token = None
+        self.pairing_code = None
+        self.pairing_code_expiration = None
         self.state = str(uuid4())
-        self.activator = None
-        self.activator_lock = Lock()
-        self.activator_cancelled = False
-
+        self.platform = None
+        self.nato_alphabet = None
         self.counter_lock = Lock()
-        self.count = -1  # for repeating pairing code. -1 = not running
-
-        self.nato_dict = None
-
         self.mycroft_ready = False
-        self.pair_dialog_lock = Lock()
-        self.paired_dialog = 'pairing.paired'
+        self.pairing_code_retry_cnt = 0
+        self.account_creation_requested = False
+
+        # These attributes track the status of the device activation
+        self.device_activation_lock = Lock()
+        self.device_activation_checker = None
+        self.device_activation_cancelled = False
+        self.activation_attempt_count = 0
+
+        # These attributes are used when tracking the ready state to control
+        # when the paired dialog is spoken.
+        self.paired_dialog_lock = Lock()
+        self.paired_dialog = None
         self.pairing_performed = False
 
-        self.num_failed_codes = 0
+        # These attributes are used when determining if pairing has started.
+        self.pairing_status_lock = Lock()
+        self.pairing_in_progress = False
 
     def initialize(self):
+        """Stuff to do after constructor but before intent."""
         self.add_event("mycroft.not.paired", self.not_paired)
-        self.nato_dict = self.translate_namedvalues('codes')
+        self.nato_alphabet = self.translate_namedvalues('codes')
+        # TODO replace self.platform logic with call to enclosure capabilities
+        self.platform = self.config_core['enclosure'].get('platform', 'unknown')
```

Comment:
pycodestyle marks this as beeing too long 80 > 79

Reply:
fixed